website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,386 @@
+
+  void drink(int liters) => state = min(state - liters, 0);
+}
+
+final wellNotifierProvider = StateNotifierProvider<WellNotifier, int>((ref) {
+  final availableToDrink = ref.watch(availableWater);
+  return WellNotifier(availableToDrink);
+});
+```
+
+Let's recap the main differences:
+
+- `StateNotifier`'s reactive dependencies are declared in its provider, whereas `Notifier`
+  centralizes this logic in its `build` method
+- Indeed, `StateNotifier`'s whole initialization process is split between its provider and its constructor,
+  whereas `Notifier` reserves a single place to place such logic
+- Notice how, as opposed to `StateNotifier`, no logic is ever written into a `Notifier`'s constructor
+
+Similar conclusions can be made with `AsyncNotifer`, `Notifier`'s asynchronous equivalent.  
+Let's check it out.
+
+## Better async support with `AsyncNotifier`
+
+The main appeal of the new API syntax is an improved DX on asynchronous states.  
+Previously, with `StateNotifier`, we would handle that like so:
+
+```dart
+class AsyncTodosNotifier extends StateNotifier<AsyncValue<List<Todo>>> {
+  AsyncTodosNotifier() : super(const AsyncLoading()) {
+    _postInit();
+  }
+
+  Future<void> _postInit() async {
+    state = await AsyncValue.guard(
+      // mock of a network request
+      () => Future.delayed(const Duration(seconds: 1), () => []),
+    );
+  }
+
+  // ...
+}
+```
+
+This heavy syntax suffers from the same defects showed above.  
+`AsyncNotifer`, just like `Notifier`, brings a simple and uniform API. Here, it's easy to see
+`AsyncNotifer` as a `FutureProvider` with methods.
+
+Here's the above example, rewritten with the new `AsyncNotifier` APIs:
+
+<AutoSnippet language="dart" {...asyncNotifier}></AutoSnippet>
+
+Therefore, migrating from `StateNotifier<AsyncValue<T>>` to a `AsyncNotifer<T>` should be very straightforward:
+
+- Place the initialization logic directly into `build`: there's no need to define a `_postInit` method anymore
+- There's no need to explicitly initialize `state` with `AsyncLoading`: as long as `build` returns a `Future`,
+  this will be handled by `AsyncNotifer`, directly
+- You can evaluate an initialization synchronously, if ever needed: just avoid returning a `Future` in your `build` method,
+  i.e. let `build` be synchronous (this is the equivalent of initializing `StateNotifier` with an `AsyncData`)
+
+Finally, `AsyncNotifer` comes with a set of utilities and getters that `StateNotifier` doesn't have, such as e.g.
+[`future`](https://pub.dev/documentation/riverpod/latest/riverpod/AutoDisposeAsyncNotifier/future.html)
+and [`update`](https://pub.dev/documentation/riverpod/latest/riverpod/AutoDisposeAsyncNotifier/update.html).
+This enables us to write much simpler logic when handling asynchronous mutations.
+
+## Explicit `.family` and `.autoDispose` modifications
+
+When a `StateNotifier` had family parameters, these were not found into its class, but they were
```

Comment:
I'd drop this first sentence entirely and focus uniquely on how notifiers have a different subclass based on the modifiers.

Reply:
We could have a small `:::info` later down talking about how family args are available with `this.argument` (or this.myVariable with codegen)